Fix port allocation race condition for elastic test

H-Huang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#65149 Fix port allocation race condition for elastic test**

Fixes #64789

There is a race condition between when the free port is acquired to when it is used to create the store in which it may have been used. Since this test only tests that timeout is triggered for tcpstore, we can bind to any port on tcpstore creation.

This only affects the test on the server (since that is where the port is used), but I changed both tests for clarity

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang @cbalioglu @gcramer23

Differential Revision: [D30993166](https://our.internmc.facebook.com/intern/diff/D30993166)

H-Huang committed:
Fix port allocation race condition for elastic test

[ghstack-poisoned]

H-Huang commented:
@H-Huang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30993166).

mrshenli reviewed (approved):
LGTM! Could you please add some comments to the test. Thanks!

H-Huang committed:
Update on "Fix port allocation race condition for elastic test"


Fixes #64789

There is a race condition between when the free port is acquired to when it is used to create the store in which it may have been used. Since this test only tests that timeout is triggered for tcpstore, we can bind to any port on tcpstore creation.

This only affects the test on the server (since that is where the port is used), but I changed both tests for clarity

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang @cbalioglu @gcramer23

Differential Revision: [D30993166](https://our.internmc.facebook.com/intern/diff/D30993166)

[ghstack-poisoned]

H-Huang commented:
@H-Huang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30993166).
